lib/active_model_serializers/model.rb

```diff
@@ -3,42 +3,50 @@
 # serializable non-activerecord objects.
 module ActiveModelSerializers
   class Model
-    include ActiveModel::Model
     include ActiveModel::Serializers::JSON
+    include ActiveModel::Model
+
+    class_attribute :attribute_names
+    self.attribute_names = []
 
     def self.attributes(*names)
       attr_accessor(*names)
+      self.attribute_names = attribute_names | names.map(&:to_sym)
     end
 
-    attr_reader :attributes, :errors
+    attributes :id
+    attr_writer :updated_at
+
+    attr_reader :errors
 
     def initialize(attributes = {})
-      @attributes = attributes && attributes.symbolize_keys
       @errors = ActiveModel::Errors.new(self)
       super
     end
 
     # Defaults to the downcased model name.
     def id
-      attributes.fetch(:id) { self.class.name.downcase }
+      @id ||= self.class.name && self.class.name.downcase
```

Comment:
The `id` method should return a unique identifier for a resource, no? Not sure if providing such a default is more useful than dangerous.